src/system/win32-wgl/interface.c

```diff
@@ -0,0 +1,239 @@
+#include <GL/gl.h>
+#include <GL/wgl.h>
+#include <GL/wglext.h>
+#include <stdio.h>
+#include <time.h>
+#include <windows.h>
+
+#include "breadbox.h"
+
+#define NSEC_PER_TICK (1000000000 / BREADBOX_TICKRATE)
+
+HWND window;
+HDC devicecontext;
+struct timespec EPOCH;
+
+// find a better way to do this
+RECT windowrect;
+int width;
+int height;
+
+LRESULT WndProc(HWND hWnd, UINT uMsg, WPARAM wParam, LPARAM lParam) {
+	switch(uMsg) {
+	case WM_DESTROY:
+		PostQuitMessage(0);
+		return 0;
+	case WM_SIZE:
+	{
+		GetClientRect(window, &windowrect);
+		width = windowrect.right - windowrect.left;
+		height = windowrect.bottom - windowrect.top;
+		
+		glViewport(0, 0, width, height);
+	}
+	break;
+	}
+	return DefWindowProc(hWnd, uMsg, wParam, lParam);
+}
+
+int create_window(char *name, int width, int height) {
+	WNDCLASS wndclass = {0};
+	wndclass.lpszClassName = "breadboxwindow";
+	wndclass.lpfnWndProc = WndProc;
+	wndclass.hbrBackground = (HBRUSH)(COLOR_BACKGROUND);
+	wndclass.style = CS_OWNDC;
+
+	RegisterClass(&wndclass);
+
+	window =
+		CreateWindow("breadboxwindow", name, WS_OVERLAPPEDWINDOW, CW_USEDEFAULT,
+					 CW_USEDEFAULT, width, height, NULL, NULL, NULL, NULL);
+
+	if(window == NULL)
+		return 1;
+
+	return 0;
+}
+
+float get_subtick() {
+    float current = 0.0;
+    struct timespec now;
+    if(clock_gettime(CLOCK_MONOTONIC, &now)) {
+        puts("breadbox_get_subtick: Failed to read monotonic clock!");
```

Comment:
This function should start with "get_subtick" rather than "breadbox_get_subtick" but I did that in my version too... I'll need to fix that.